ext/AvizExt/viz/clustering.jl

```diff
@@ -118,52 +155,66 @@ function ADRIA.viz.map!(
-
+    _get_colors(clusters::Vector{Int64}, alphas::Vector{Float64})::Vector{RGBA{Float32}}
+    _get_colors(clusters::Vector{Int64}, alpha::Float64)::Vector{RGBA{Float32}}
+    _get_colors(clusters::Vector{Int64})::Vector{RGBA{Float32}}
 Vector of cluster colors.
 
 # Arguments
 - `clusters` : Vector of numbers corresponding to clusters
+- `alphas` : Vector of alphas to be used for cluster color
 
 # Returns
 Vector{RGBA{Float32}}
 """
-function _clusters_colors(clusters::Vector{Int64})::Vector{RGBA{Float32}}
+function _get_colors(clusters::Vector{Int64}, alphas::Vector{Float64})::Vector{RGBA{Float32}}
     # Number of non-zero clusters
     n_clusters = length(unique(filter(cluster -> cluster != 0, clusters)))
 
     # Vector of clusters colors for non-zero clusters
-    clusters_colors = categorical_colors(:seaborn_bright, n_clusters)
+    colors = categorical_colors(:seaborn_bright, n_clusters)
+
+    # Apply alpha to cluster colors
+    if !isempty(alphas)
+        colors = [RGBA(c.r, c.g, c.b, a) for (c, a) in zip(colors, alphas)]
+    end
 
     # Assign color "black" to cluster == 0
     rgba_black = parse(RGBA{Float32}, "transparent")
-    return [cluster == 0 ? rgba_black : clusters_colors[cluster] for cluster in clusters]
+    return [cluster == 0 ? rgba_black : colors[cluster] for cluster in clusters]
+end
+function _get_colors(clusters::Vector{Int64}, alpha::Float64)::Vector{RGBA{Float32}}
+    alphas::Vector{Float64} = fill(alpha, length(clusters))
+    return _get_colors(clusters, alphas)
+end
+function _get_colors(clusters::Vector{Int64})::Vector{RGBA{Float32}}
+    return _get_colors(clusters, 1.0)
 end
 
 """
-    _cluster_color(unique_cluster_colors::Vector{RGBA{Float32}}, clusters::Vector{Int64}, cluster::Int64)
+    _get_alphas(clusters::Vector{Int64})::Vector{Float64}
 
-Color parameter for current cluster weighted by number of scenarios
+Vector of color alphas for each clusters weighted by number of scenarios
 
 # Arguments
-- `unique_cluster_colors` : Vector of all cluster options that are being used
-- `clusters` : Vector of numbers corresponding to clusters
-- `cluster` : Current cluster
+- `clusters` : Vector with scenario cluster numbers
 
 # Returns
-Tuple{RGBA{Float32}, Float64}
+Vector with one color alpha for each cluster
 """
-function _cluster_color(unique_cluster_colors::Vector{RGBA{Float32}},
-    clusters::Vector{Int64}, cluster::Int64)::Tuple{RGBA{Float32},Float64}
-    # Number of scenarios for a cluster
-    n_scens = count(clusters .== cluster)
+function _get_alphas(clusters::Vector{Int64})::Vector{Float64}
+    alphas::Vector{Float64} = zeros(Float64, length(unique(clusters)))
 
-    # Compute line weight
-    color_weight = max(min((1.0 / (n_scens * 0.05)), 0.6), 0.1)
+    for (i, cluster) in enumerate(unique(clusters))
+        n_scens = count(clusters .== cluster)
+        base_alpha = 1.0 / (n_scens * 0.05)
+        alphas[i] = max(min(base_alpha, 0.6), 0.1)
+    end

```

Comment:
Thinking about this now, shouldn't the weighting be based on total number of scenarios?
What we have right now could lead to a smaller cluster hiding a larger cluster (unless I'm misinterpreting the code here?)

Reply:
I don't think a small cluster would hide a large one because alpha is, at the most, 0.6 (for small clusters) but in this case the cluster has only a few lines.

Considering the result (figure above) it seems to me that this works fine. 

That said, if I would do this now, I would probably try something like:

```julia
base_alpha = 1 - count(clusters .== cluster)/length(clusters)
alphas[i] = max(min(base_alpha, 0.6), 0.1)
```

Reply:
But I think this change is out of the scope of this PR so if you think we should change this I suggest we open an issue, what do you think?

Reply:
No need for an issue if you're going to submit a PR directly after (but please do if your not planning to do it immediately).

By hiding, take a look at the example figures provided here. See how the orange trajectories are blocked from view.

https://github.com/open-AIMS/ADRIA.jl/pull/492#issue-1902179842

Maybe we plot smaller clusters last?

Reply:
You wrote "smaller cluster hiding a larger cluster" but in this figure what I see is a larger cluster hiding a smaller cluster, right? But yes, I think a good improvement would be to plot clusters in order from larger to smaller. 

Let me open a different PR to do that and also try to change the way this alpha is computed and test some different cases.

Reply:
> You wrote "smaller cluster hiding a larger cluster" but in this figure what I see is a larger cluster hiding a smaller cluster, right?

Yes, correct, accidentally swapped those, sorry for the miscommunication

Reply:
Maybe the problem with the figures you sent from PR 492 wasn't the alpha weight.

If you see my last commit - and the commit message - you will see what happened:

Currently we are iterating over the cluster with a `for cluster in unique(clusters)` but if clusters vector is something like `[2, 1, 1, 3, 2]`, then `unique(clusters)` will return `[2, 1, 3]`, and then cluster number `2`, for instance, will get the weight that should be assigned to cluster `1`, because we are using the index of the cluster inside the vector as it was the number of the cluster itself.

Reply:
Update: I just tested here and that doesn't solve the visualization problem. I think plotting larger clusters before smaller will solver though.